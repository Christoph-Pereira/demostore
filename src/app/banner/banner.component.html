<div class="banner">
    <p class="bannerText">
        This is so beautiful I am going to blame someone else
    </p>
</div>