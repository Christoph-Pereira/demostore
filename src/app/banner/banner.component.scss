.banner {
    background-image: url("../../assets/images/Banner.png");
    min-height: 35.5rem;
}
.bannerText {
    color: white;
    font-size: 2rem;
    padding: 12rem 8rem;
}