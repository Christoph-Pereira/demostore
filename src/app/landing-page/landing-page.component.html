<app-banner></app-banner>
<app-quick-links></app-quick-links>
<app-product-links></app-product-links>
