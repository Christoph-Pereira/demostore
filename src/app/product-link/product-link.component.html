<a routerLink="{{url}}">
    <div class="productLinkDiv" [ngStyle]="{backgroundImage: 'url(' + image + ')'}">
        <p class="productLinkText">
            {{text}}
        </p>
    </div>
</a>