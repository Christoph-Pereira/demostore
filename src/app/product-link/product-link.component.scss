a {
    text-decoration: none;
}
.productLinkDiv {
    height: 450px;
    width: 685px;
    margin: auto;
    transition: all .2s ease-in-out;

    .productLinkText {
        font-size: 2rem;
        padding: 2rem;
        color: white;
    }
}

.productLinkDiv:hover {
    transform: scale(1.1);
}