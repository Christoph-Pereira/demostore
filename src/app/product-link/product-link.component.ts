import { Component, Input, OnInit } from '@angular/core';

@Component({
  selector: 'app-product-link',
  templateUrl: './product-link.component.html',
  styleUrls: ['./product-link.component.scss']
})
export class ProductLinkComponent {
  
  @Input() public text: String | undefined;
  @Input() public image: String | undefined;
  @Input() public url: String | undefined;

  constructor() { }

}
