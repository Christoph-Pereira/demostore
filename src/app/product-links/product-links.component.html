<div class="productLinksDiv">
    <h1>Showcased Solutions</h1>
    <div class ="productLinks">
        <div *ngFor="let productLink of productLinks">
            <app-product-link [text]="productLink.text" [image]="productLink.image" [url]="productLink.url"></app-product-link>
        </div>
    </div>
</div>