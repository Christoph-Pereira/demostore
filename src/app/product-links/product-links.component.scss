.productLinksDiv {
    align-items: center;
    color: white;
    padding: 2rem;

    h1 {
        text-align: center;
        color: black;
        margin: 2rem;
        font-size: 2rem;

    }
    .productLinks {
        display: grid;
        grid-template-columns: auto auto;
    }
}