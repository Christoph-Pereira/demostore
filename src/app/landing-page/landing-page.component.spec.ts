import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ProductLinkComponent } from '../product-link/product-link.component';

import { LandingPageComponent } from './landing-page.component';
import { BannerComponent } from '../banner/banner.component';
import { QuickLinksComponent } from '../quick-links/quick-links.component';
import { ProductLinksComponent } from '../product-links/product-links.component';
import { MockComponents } from 'ng-mocks';

describe('LandingPageComponent', () => {
  let component: LandingPageComponent;
  let fixture: ComponentFixture<LandingPageComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ LandingPageComponent, MockComponents (BannerComponent, QuickLinksComponent, ProductLinksComponent) ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(LandingPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});

