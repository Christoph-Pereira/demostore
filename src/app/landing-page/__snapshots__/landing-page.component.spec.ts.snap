// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPageComponent should match snapshot 1`] = `
<app-landing-page
  __ngContext__={[Function LRootView]}
>
  <app-banner /><app-quick-links /><app-product-links />
</app-landing-page>
`;
